docs/spec_draft.md

```diff
@@ -1244,6 +1245,68 @@ produces an implementation-defined value.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.dynamic_update_slice
+
+### Semantics
+
+Produces a `result` tensor which is equal to the `operand` tensor except that
+the slice starting at `start_indices` is updated with the values in `update`.
+
+More formally, `result[i0, ..., iR-1]` is defined as:
+  * `update[j0, ..., jR-1]` if `jd = adjusted_start_indices[d][] + id` where
+    `adjusted_start_indices = clamp(0, start_indices, shape(operand) - update)`.
+  * `operand[i0, ..., iR-1]` otherwise.
+
+### Inputs
+
+| Name            | Type                                                    |
+|-----------------|---------------------------------------------------------|
+| `operand`       | tensor of any supported type                            |
+| `update`        | tensor of any supported type                            |
+| `start_indices` | variadic number of 0-dimensional tensors of type `si64` |
+
+### Outputs
+
+| Name     | Type                         |
+|----------|------------------------------|
+| `result` | tensor of any supported type |
+
+### Constraints
+
+  * (C1) `operand` and `result` have the same type.
+  * (C2) element_type(`update`) $=$ element_type(`operand`).
+  * (C3) rank(`update`) $=$ rank(`operand`).
+  * (C4) size(`start_indices`) $=$ rank(`operand`).
```

Comment:
Similarly to what's proposed in #556, let's add " * (C5) All `start_indices` have the same type".